src/user/mod.rs

```diff
@@ -378,6 +378,24 @@ pub fn updates(req: &mut Request) -> CargoResult<Response> {
     Ok(req.json(&R{ versions: versions, meta: Meta { more: more } }))
 }
 
+/// Handles the `GET /users/:user_id/stats` route.
+pub fn stats(req: &mut Request) -> CargoResult<Response> {
+    use  diesel::expression::dsl::sum;
+    let name = &req.params()["user_id"].parse::<i32>().ok().unwrap();
+    let conn = req.db_conn()?;
+    
+    let data = crate_downloads::table.filter(crate_downloads::crate_id.eq_any(
+        crate_owners::table.select(crate_owners::crate_id)
+            .filter(crate_owners::owner_id.eq(name))
+        )).select(sum(crate_downloads::downloads)).first::<i64>(&*conn)?; 
```

Comment:
@sgrif reg `user_id` - Ahhh! You're right. I did not realise that before :) pushed a commit now to restrict that.

regarding `joinable!` - raised an issue on diesel to get some doubts cleared - https://github.com/diesel-rs/diesel/issues/938